### PR TITLE
[apiConformance] Add test CheckInferIsNotChangeInput to io_tensor

### PR DESCRIPTION
**Pull request by sbalandi**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *CVS-111339*

### CODE REVIEW

**Comment by iefode** [comment]
@ilya-lavrenov Could you please take a look?

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 1]
```diff
@@ -219,6 +219,23 @@ TEST_P(OVInferRequestIOTensorTest, InferStaticNetworkSetChangedOutputTensorThrow
...
+
+    ov::Tensor input_after_infer;
+    OV_ASSERT_NO_THROW(input_after_infer = req.get_tensor(input));
+    ASSERT_EQ(input_tensor.data(), input_after_infer.data());
+
+    OV_ASSERT_NO_THROW(req.infer());
+
+    ov::Tensor input_after_several_infer;
+    OV_ASSERT_NO_THROW(input_after_several_infer = req.get_tensor(input));
+    ASSERT_EQ(input_tensor.data(), input_after_several_infer.data());
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:236]: can we test the data itself?
Here we check that pointer is not reallocated, but we need to check that values are the same as well

Comment (sbalandi) [reply src/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:238]: yes, comparison of values is added

Comment (ilya-lavrenov) [reply src/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:238]: I meant check that `data()` contains the same values

Comment (sbalandi) [reply src/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp]: moved to comparison with ov::test::utils::compare , it is checked values in this function

**Review by ilya-lavrenov — APPROVED** [review]

**Review by andrei-kochin — APPROVED** [review]

**Pull request merged**